mypy/build.py

```diff
@@ -897,12 +898,26 @@ def write_cache(id: str, path: str, tree: MypyFile,
         data_mtime = os.path.getmtime(data_json)
         manager.trace("Interface for {} is unchanged".format(id))
     else:
-        with open(data_json_tmp, 'w') as f:
-            f.write(data_str)
-            f.write('\n')
-        data_mtime = os.path.getmtime(data_json_tmp)
-        os.replace(data_json_tmp, data_json)
         manager.trace("Interface for {} has changed".format(id))
+        try:
```

Comment:
It's unfortunate that the error handling (including comments) is longer than the main logic of the function, but I'm hesitant about breaking it up into smaller bits. Maybe it'll fit on your screen if I move the comments elsewhere (e.g. to the function's docstring)? Anyway, we can iterate on that after the merge.